files/nagios/plugins/check_openstack_interface.py

```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/env python3
+
+# Copyright (C) 2021 Canonical Ltd.
+
+# Authors:
+#   Robert Gildein <robert.gildein@canonical.com>
+
+import argparse
+from configparser import ConfigParser
+
+import openstack
+from nagios_plugin3 import try_check, UnknownError, CriticalError
+
+SEPARATOR = ","
+ACTIVE = "ACTIVE"
+DOWN = "DOWN"
+VALID_STATUS = [ACTIVE, DOWN]
+INTERFACE = {
+    "network": "list_networks",
+    "floating-ip": "list_floating_ips",
+    "server": "list_servers",
+    "port": "list_ports",
+    "security-group": "list_security_groups",
+    "subnet": "list_subnets",
+}
+INTERFACE_BY_EXISTENCE = ["security-group", "subnet"]
+
+
+def _parse_arguments():
+    """Parse the check arguments and connect to OpenStack.
+
+    :returns: credentials, interface name, set IDs,
+              set IDs to skip, values to filter when using `--all`
+              and check all flag
+    :rtype: Tuple[configparser.ConfigParser, str, set, set, dict, bool]
+    """
+    credentials = ConfigParser()
+    parser = argparse.ArgumentParser("check_openstack_interface")
+    parser.add_argument("interface", type=str, help="interface name")
+    parser.add_argument("-c", "--credential", required=True,
+                        type=argparse.FileType("r"),
+                        help="path to OpenStack credential cnf file")
+    parser.add_argument("--all", action="store_true", help="check all")
+    parser.add_argument("-i", "--id", action="append", type=str, default=[],
+                        help="check specific id")
+    parser.add_argument("--skip-id", action="append", type=str, default=[],
+                        help="skip specific id")
+    parser.add_argument("--select", action="append", type=str, default=[],
+                        help="use `--select` together with `--all`"
+                             "(e.g. --select subnet=<id>)")
+    args = parser.parse_args()
+
+    if args.interface not in INTERFACE:
+        parser.error("`{}` interface is not supported".format(args.interface))
+
+    if args.all and args.interface in INTERFACE_BY_EXISTENCE:
+        parser.error("flag `--all` is not supported with "
+                     "interface {}".format(args.interface))
+    if args.all and args.id:
+        parser.error("`--all` and `--id` couldn't be given together")
```

Comment:
what about `'--all/--id' are mutually exclusive`

Reply:
If the above does not sound good, I'd suggest replacing "couldn't" by "can't" ("can't be used together").